models/forms/PublishProposalForm.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+
+namespace app\models\forms;
+
+
+use yii\base\Model;
+
+class PublishProposalForm extends Model
+{
+    const TWEET_MAX_CHARS = 280;
+    public $content;
+    public $file;
+    public $social_media;
+
+    public function rules()
+    {
+        return [
+            [['content', 'social_media', 'file'], 'required'],
+            ['content', 'validateContent'],
+        ];
+    }
+
+    public function validateContent($attribute)
+    {
+        if (mb_strlen($this->content) > self::TWEET_MAX_CHARS) {
+            $this->addError($attribute, 'Too long content');
```

Comment:
For a better UX, you should precise the max content length in the error message

Reply:
Yes, more clearly for users